File: app/src/main/java/vinhlong/ditagis/com/khaosatdongho/utities/Constant.java
package vinhlong.ditagis.com.khaosatdongho.utities;

import java.text.SimpleDateFormat;
import java.util.Arrays;
import java.util.List;

/**
 * Created by ThanLe on 3/1/2018.
 */

public class Constant {
    public static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("dd/MM/yyyy");
    public static final String DIACHI = "DiaChi";
    public static final String EMPTY = "";
    public static final String NGAY_CAP_NHAT = "NgayCapNhat";
    public static final int REQUEST_LOGIN = 0;
    private static final String SERVER_API = "http://vwa.ditagis.com/api";


    //    private static final String SERVER_API = "http://113.161.88.180:798/apiv1/api";
    public class AttachmentName {
        public static final String ADD = "img_%s_%d.png";
        public static final String UPDATE = "img_%s_%d.png";
    }

    public static class API_URL {
        public static final String LOGIN = SERVER_API + "/Login";
        public static final String DISPLAY_NAME = SERVER_API + "/Account/Profile";
        public static final String LAYER_INFO = SERVER_API + "/Account/LayerInfo";
        public static final String IS_ACCESS = SERVER_API + "/Account/IsAccess/m_gmdhks";
        public static final String TENMAU_LIST = SERVER_API + "/gmdh/thietlapmau/laytenmau";
        public static final String VATTU_THEOMAU = SERVER_API + "/odata/gmdh/ThietLapMaus?$select=*&$filter=TENMAU eq '%s'";
        public static final String VATTU_LIST = SERVER_API + "/odata/GMDH/VatTuThietKes";
        public static final String INSERT_VATTU = SERVER_API + "/odata/gmdh/HoSoGanMoiVatTuTKs";
        public static final String DELETE_VATTUS = SERVER_API + "/GMDH/HoSoGanMoiVatTu/XoaVatTuTheoIDKhachHang/%s";
        public static final String VATTUS_DONGHO = SERVER_API + "/odata/GMDH/HoSoGanMoiVatTuTKs?$select=*&$filter=IDKhachHang eq '%s'";
    }
    public static class METHOD {
        public static final String DELETE = "DELETE";
        public static final String POST = "POST";
        public static final String GET = "GET";
    }
    public static class REQUEST {
        public static final int ID_UPDATE_ATTACHMENT = 50;
        public static final int ID_UPDATE_ATTRIBUTE = 10;
    }
    public static class TENMAU {
        public static final String SELECT = "Chọn thiết lập mẫu";
    }
    public static class LayerFields {
        public static final String OBJECTID = "OBJECTID";
    }

    public static class DefineST {
        public static final String DonViTien = " (đồng)";
    }

    public static class TinhTrangDongHoKhachHang {
        public static final String DANG_KHAO_SAT = "DKS";
        public static final String DANG_THIET_KE = "DTK";
    }
    public static class DongHoKhachHangFields {
        public static final String OBJECT_ID = "OBJECTID";
        public static final String ID = "ID";
        public static final String TINH_TRANG = "TinhTrang";
        public static final String CMND = "CMND";
        public static final String TEN_KH = "TenKH";
        public static final String GHI_CHU = "GhiChu";
        public static final String GHI_CHU_KS = "GhiChuKS";
        public static final String NGAY_CAP_NHAT = "TGGiaoKS";
        public static final String NGUOI_CAP_NHAT = "NVKhaoSat";
        public static final String SO_DIEN_THOAI = "SoDienThoai";
        public static final String DIA_CHI = "DiaChi";
        public static final String DIA_CHI_LAP_DAT = "DiaChiLapDat";

        public static final List<String> UpdateFields = Arrays.asList(TEN_KH, CMND, DIA_CHI_LAP_DAT, SO_DIEN_THOAI, GHI_CHU_KS);
        public static final List<String> OutFields = Arrays.asList(ID, TEN_KH, CMND, DIA_CHI, DIA_CHI_LAP_DAT, SO_DIEN_THOAI, GHI_CHU_KS);
    }


    public static class VatTuFields {
        public static final String DBDongHo = "DBDongHo";
        public static final String MaKhachHang = "CMND";
        public static final String MaVatTu = "MaVatTu";
        public static final String SoLuong = "SoLuong";
        public static final String NgayCapNhat = "NgayCapNhat";
    }

    public static class VatTuAlias {
        public static final String LoaiVatTu = "Loại vật tư";
    }

    public static class LoaiVatTuFields {
        public static final String MaVatTu = "MaVatTu";
        public static final String TenVatTu = "TenVatTu";
        public static final String DonViTinh = "DonViTinh";
        public static final String GiaVatTu = "VT";
        public static final String GiaNC = "GiaNC";
        public static final String GiaMay = "GiaMay";
    }
    public static class HanhChinhFields {
        public static final String MAXA = "MAXA";
        public static final String TENHANHCHINH = "TenHanhChinh";
        public static final String MAHUYEN = "MAHUYEN";
        public static final String TENHUYEN = "TenHuyen";
    }
    public static final int idDongHoKhachHang = 5;

    public static class IDLayer {
        public static final String BASEMAP = "BASEMAP";
        public static final String CHUYENDE = "CHUYENDE";
        public static final String DHKHLYR = "gmdhLYR";
        public static final String VATTUDONGHOTBL = "vattudonghoTBL";
        public static final String DMVATTUTBL = "vattuthietkeBTL";

    }

    private static Constant mInstance = null;

    public static Constant getInstance() {
        if (mInstance == null)
            mInstance = new Constant();
        return mInstance;
    }

    private Constant() {

    }

}

File: app/src/main/java/vinhlong/ditagis/com/khaosatdongho/utities/Constant.kt
package vinhlong.ditagis.com.khaosatdongho.utities

import java.text.SimpleDateFormat
import java.util.Arrays

/**
 * Created by ThanLe on 3/1/2018.
 */

class Constant private constructor() {


    //    private static final String SERVER_API = "http://113.161.88.180:798/apiv1/api";
    object AttachmentName {
        val ADD = "img_%s_%d.png"
        val UPDATE = "img_%s_%d.png"
    }
    object PreferenceKey {
        const val USERNAME = "hovaten"
        const val FULL_NAME = "ten"
        const val EMAIL = "email"
        const val PASSWORD = "sdt"
        const val USER_MANUAL = "user_manual"
        const val LOGIN_API = "preference_login_api"
    }
    object API_URL {
        val LOGIN = "$SERVER_API/Login"
        val DISPLAY_NAME = "$SERVER_API/Account/Profile"
        val LAYER_INFO = "$SERVER_API/Account/LayerInfo"
        val IS_ACCESS = "$SERVER_API/Account/IsAccess/m_gmdhks"
        val TENMAU_LIST = "$SERVER_API/gmdh/thietlapmau/laytenmau"
        val VATTU_THEOMAU = "$SERVER_API/odata/gmdh/ThietLapMaus?\$select=*&\$filter=TENMAU eq '%s'"
        val VATTU_LIST = "$SERVER_API/odata/GMDH/VatTuThietKes"
        val INSERT_VATTU = "$SERVER_API/odata/gmdh/HoSoGanMoiVatTuTKs"
        val DELETE_VATTUS = "$SERVER_API/GMDH/HoSoGanMoiVatTu/XoaVatTuTheoIDKhachHang/%s"
        val VATTUS_DONGHO = "$SERVER_API/odata/GMDH/HoSoGanMoiVatTuTKs?\$select=*&\$filter=IDKhachHang eq %d"
    }

    object METHOD {
        val DELETE = "DELETE"
        val POST = "POST"
        val GET = "GET"
    }

    object REQUEST {
        val ID_UPDATE_ATTACHMENT = 50
        val ID_UPDATE_ATTRIBUTE = 10
        val ID_UPDATE_VAT_TU = 11
    }

    object TENMAU {
        val SELECT = "Chọn thiết lập mẫu"
    }

    object LayerFields {
        val OBJECTID = "OBJECTID"
    }

    object DefineST {
        val DonViTien = " (đồng)"
    }

    object TinhTrangDongHoKhachHang {
        val DANG_KHAO_SAT = "DKS"
        val DANG_THIET_KE = "DTK"
    }

    object DongHoKhachHangFields {
        val OBJECT_ID = "OBJECTID"
        val ID = "ID"
        val TINH_TRANG = "TinhTrang"
        val CMND = "CMND"
        val TEN_KH = "TenKH"
        val GHI_CHU = "GhiChu"
        val GHI_CHU_KS = "GhiChuKS"
        val NGAY_CAP_NHAT = "TGGiaoKS"
        val NGUOI_CAP_NHAT = "NVKhaoSat"
        val SO_DIEN_THOAI = "SoDienThoai"
        val DIA_CHI = "DiaChi"
        val DIA_CHI_LAP_DAT = "DiaChiLapDat"

        val UpdateFields = Arrays.asList(TEN_KH, CMND, DIA_CHI_LAP_DAT, SO_DIEN_THOAI, GHI_CHU_KS)
        val OutFields = Arrays.asList(ID, TEN_KH, CMND, DIA_CHI, DIA_CHI_LAP_DAT, SO_DIEN_THOAI, GHI_CHU_KS)
    }


    object VatTuFields {
        val DBDongHo = "DBDongHo"
        val MaKhachHang = "CMND"
        val ID = "IDKhachHang"
        val MaVatTu = "MaVatTu"
        val SoLuong = "SoLuong"
        val NgayCapNhat = "NgayCapNhat"
        val GIA_NC = "GiaNC"
    }

    object VatTuAlias {
        val LoaiVatTu = "Loại vật tư"
    }

    object LoaiVatTuFields {
        val MaVatTu = "MaVatTu"
        val TenVatTu = "TenVatTu"
        val DonViTinh = "DonViTinh"
        val GiaVatTu = "VT"
        val GiaNC = "GiaNC"
        val GiaMay = "GiaMay"
    }

    object HanhChinhFields {
        val MAXA = "MAXA"
        val TENHANHCHINH = "TenHanhChinh"
        val MAHUYEN = "MAHUYEN"
        val TENHUYEN = "TenHuyen"
    }

    object IDLayer {
        val BASEMAP = "BASEMAP"
        val CHUYENDE = "CHUYENDE"
        val DHKHLYR = "gmdhLYR"
        val VATTUDONGHOTBL = "vattudonghoTBL"
        val DMVATTUTBL = "vattuthietkeBTL"

    }

    companion object {
        val DATE_FORMAT = SimpleDateFormat("dd/MM/yyyy")
        val DIACHI = "DiaChi"
        val EMPTY = ""
        val NGAY_CAP_NHAT = "NgayCapNhat"
        val PATH = "vlong_ksatdongho"
        val REQUEST_LOGIN = 0
        private val SERVER_API = "http://vwa.ditagis.com/api"
        val idDongHoKhachHang = 5

        private var mInstance: Constant? = null

        val instance: Constant
            get() {
                if (mInstance == null)
                    mInstance = Constant()
                return mInstance!!
            }
    }

}
